### ISSUE TITLE
Why mapping of QueryProc<T> result is case sensitive?

### ISSUE THREAD

**Issue by feeleen** [linq2db/linq2db#2088]
Why mapping of QueryProc<T> result is case sensitive?
I think its a bug..
If procedue returns "id", and my data model class has property "ID" - mapper won't fill this property. (((
I use sql server.

**Comment by sdanyliv** [comment]
@feeleen, try to change comparer for `MappingSchema`. By default it is `StringComparer.Ordinal`

Global
```cs
MappingSchema.Default.ColumnNameComparer = StringComparer.OrdinalIgnoreCase
```
Or for specific connection
```cs
var ms = new MappingSchema();
ms.ColumnNameComparer = StringComparer.OrdinalIgnoreCase;

// through constructor
using var db = new DataConnection(ms);

// or via method
db.AddMappingSchema(ms);
```

**Comment by feeleen** [comment]
Thank you, setting MappingSchema.Default.ColumnNameComparer to 
StringComparer.OrdinalIgnoreCase works!

**Comment by MaceWindu** [comment]
I'll reopen it as we still need to fix it on linq2db side I think. Will check later

**Comment by feeleen** [comment]
Yes, it seems a bug there...
If I query via table - mapping works fine (case insensitive), if I query procedure - mapping becomes case sensitive. Just checked v2.9.5

**Comment by sdanyliv** [comment]
It is because in query variant we do not need column name. linq2db uses column index for reading property.
But yes it possible bug, SQL Server is case insensitive in SQL queries.

**Comment by feeleen** [comment]
But how do u get column indexes without comparing names? My model class doesn't contain all columns of db table.. (only few of them). And my data model class know nothing about column indexes.. 
So i do not understand your notice about indexes..

**Comment by sdanyliv** [comment]
Nevermind, it is internal realization. If you interested in, i can explain later. Better to fix/implement something important.

**Comment by feeleen** [comment]
Tried to debug & investigate the code, found some strange things:

1. SqlServerMappingSchema doesn't set it's ColumnNameComparer to StringComparer.OrdinalIgnoreCase (but I think it should!)
As a result, when ColumnNameComparer is not initialized by scheme itself, then it's going to be always Ordinal..

![image](https://user-images.githubusercontent.com/16134154/74590187-213ad980-501d-11ea-8e05-2a8f130f41aa.png)

I suggest we shoud add ColumnNameComparer initialization to SqlServerMappingSchema constructor

2. Why do I see additional connection string (and additional schema and configuration also) pointing to aspnetdb.mdf on a LocalSqlServer? Probably this comes from machine.config, but why load this connection by default? I see potential problems here if we'll set ColumnNameComparer in
 SqlServerMappingSchema and then try to connect to mySql database - ColumnNameComparer would be taken from the first schema (schema[0]) - and it is actually aspnetdb.mdf on a LocalSqlServer..
![image](https://user-images.githubusercontent.com/16134154/74590108-77f3e380-501c-11ea-96f4-0727dfd3840c.png)

**Comment by feeleen** [comment]
I think ColumnComparer shoulldn't always take shema[0]. It should take associated schema..

**Comment by MaceWindu** [comment]
Closed by #2100
